### DISCUSSION TITLE
R Memory Management

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#795] [Q&A] [answered]
Dear Tim Algoritma

Saya mendapatkan permasalahan pada saat saya menjalankan syntax ini:

<img width="558" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/12cbad34-b15f-4385-9594-cd8f878074e6">

data saya jumlah observasinya 121.654, dari hasil cek sessionInfo() seperti ini:
<img width="607" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/64467f89-f64f-40da-b96a-7c8a96bd84fe">

<img width="598" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/422cf147-3cb5-4b66-8936-36e0208b282f">

Apakah ada solusinya untuk permasalahan jika mendapatkan kondisi yang seperti ini ? Mohon sarannya.

Best regards,
JT

**Comment by arkanayudhistira** [comment, marked as answer]
Dear @joejnt,

Sepertinya error tersebut disebabkan oleh data digunakan yang terlalu besar (menggunakan memory hingga 55 Gb), dimana kemungkinan lebih besar daripada memory yang tersedia pada komputer Bapak. Untuk mengecek berapa banyak memory yang tersedia untuk digunakan, silahkan cek Memory Usage Report pada bagian panel Environment disini:
![image](https://github.com/teamalgoritma/community/assets/100895120/8755c88c-79d8-47c8-a482-f4f8ff4c63f4)

Maka akan terlihat berapa besar memory yang tersedia di komputer Bapak dan yang tersedia untuk digunakan pada RStudio
![image](https://github.com/teamalgoritma/community/assets/100895120/3540cfa7-11df-49b3-bf7a-20b2fd1f2712)

Solusi yang dapat dilakukan adalah mengurangi jumlah data pada dataset yang Bapak gunakan, agar memory yang diperlukan lebih rendah.

Semoga membantu!

**Comment by joejnt** [comment]
Terima kasih buat informasinya,

Best regards,
JT
